libnavigation-core/src/test/java/com/mapbox/navigation/core/internal/utils/RouterExTest.kt

```diff
@@ -0,0 +1,30 @@
+package com.mapbox.navigation.core.internal.utils
+
+import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.route.internal.RouterWrapper
+import io.mockk.mockk
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class RouterExTest {
+
+    @Test
+    fun testIsInternalImplementation() {
+        listOf<Triple<String, Router, Boolean>>(
+            Triple(
+                "Router interface",
+                mockk(),
+                false
+            ),
+            Triple(
+                "RouterWrapper, internal implementation",
+                mockk<RouterWrapper>(),
+                true
+            ),
+        ).let { testCases ->
+            testCases.forEach { (message, router, isInternal) ->
+                assertEquals(message, isInternal, router.isInternalImplementation())
+            }
+        }
+    }
+}
```

Comment:
I think you will get more value from the test which interacts with `MapboxNavigation` and verifies which router is passed to NN when user provides or doesn't provides custom router. 

BTW, maybe we should add instead an instrumentation test for custom router? How do we know if it works?

Reply:
> I think you will get more value from the test which interacts with MapboxNavigation and verifies which router is passed to NN when user provides or doesn't provides custom router.

for the unit test, it should be enough to verify that SDK gets the right implementation I think. 
For providing Router SDK consumes modules library, not sure that SDK should test outer library. 

> BTW, maybe we should add instead an instrumentation test for custom router? How do we know if it works?

I'm not sure that we have a good option to test custom router for now as we cannot force NN re-request route or refresh route. it might be a good option when we migrate both `Reroute` and `Refresh route` to NN. I described my concerns in the head https://github.com/mapbox/mapbox-navigation-android/pull/5500#issue-1148114117 (see the very last point). 
For now, I only added `NavigatorTest` that verify SDK serializes data for NN right

Reply:
You can provide a custom router with hardcoded `DirectionsResponse` instead of providing it via mock http server in test. Try to use navigation. If navigation successfully completes then custom router works. What do you think about it?

Reply:
> If navigation successfully completes then custom router works.
Navigation has refresh/reroute/alternative route API, we need to check that as well. But it's a good idea, I like it 👍